roles/openshift_metrics/tasks/generate_hawkular_certificates.yaml

```diff
@@ -0,0 +1,227 @@
+    hostnames: "hawkular-metrics,{{ openshift_metrics_hawkular_metrics_hostname }}"
+- name: generate hawkular-cassandra certificates
+  include: setup_certificate.yaml
+  vars:
+    component: hawkular-cassandra
+    hostnames: hawkular-cassandra
+- name: check existing aliases on the hawkular-cassandra truststore
+  shell: >
+    keytool -noprompt -list
+    -keystore {{ openshift_metrics_certs_dir }}/hawkular-cassandra.truststore
+    -storepass "$(<
+    '{{ openshift_metrics_certs_dir }}/hawkular-cassandra-truststore.pwd')"
+    | sed -n '7~2s/,.*$//p'
+  register: hawkular_cassandra_truststore_aliases
+  changed_when: false
+- name: check existing aliases on the hawkular-metrics truststore
+  shell: >
+    keytool -noprompt -list
+    -keystore {{ openshift_metrics_certs_dir }}/hawkular-metrics.truststore
+    -storepass "$(<
+    '{{ openshift_metrics_certs_dir }}/hawkular-metrics-truststore.pwd')"
+    | sed -n '7~2s/,.*$//p'
+  register: hawkular_metrics_truststore_aliases
+  changed_when: false
+- name: import the hawkular metrics cert into the cassandra truststore
+  shell: >
+    keytool -noprompt -import -v -trustcacerts
+    -alias hawkular-metrics
+    -file '{{ openshift_metrics_certs_dir }}/hawkular-metrics.crt'
+    -keystore '{{ openshift_metrics_certs_dir }}/hawkular-cassandra.truststore'
+    -storepass "$(<
+    '{{ openshift_metrics_certs_dir }}/hawkular-cassandra-truststore.pwd')"
+  when: >
+    'hawkular-metrics' not in
+    hawkular_cassandra_truststore_aliases.stdout_lines
+- name: import the hawkular cassandra cert into the hawkular metrics truststore
+  shell: >
+    keytool -noprompt -import -v -trustcacerts
+    -alias hawkular-cassandra
+    -file '{{ openshift_metrics_certs_dir }}/hawkular-cassandra.crt'
+    -keystore '{{ openshift_metrics_certs_dir }}/hawkular-metrics.truststore'
+    -storepass "$(<
+    '{{ openshift_metrics_certs_dir }}/hawkular-metrics-truststore.pwd')"
+  when: >
+    'hawkular-cassandra' not in
+    hawkular_metrics_truststore_aliases.stdout_lines
+- name: import the hawkular cassandra cert into the cassandra truststore
+  shell: >
+    keytool -noprompt -import -v -trustcacerts
+    -alias hawkular-cassandra
+    -file '{{ openshift_metrics_certs_dir }}/hawkular-cassandra.crt'
+    -keystore '{{ openshift_metrics_certs_dir }}/hawkular-cassandra.truststore'
+    -storepass "$(<
+    '{{ openshift_metrics_certs_dir }}/hawkular-cassandra-truststore.pwd')"
+  when: >
+    'hawkular-cassandra' not in
+    hawkular_cassandra_truststore_aliases.stdout_lines
+- name: import the ca certificate into the cassandra truststore
+  shell: >
+    keytool -noprompt -import -v -trustcacerts
+    -alias '{{ item }}'
+    -file '{{ openshift_metrics_certs_dir }}/ca.crt'
+    -keystore '{{ openshift_metrics_certs_dir }}/hawkular-cassandra.truststore'
+    -storepass "$(<
+    '{{ openshift_metrics_certs_dir }}/hawkular-cassandra-truststore.pwd')"
+  with_items:
+  - ca
+  - metricca
```

Comment:
I don't think this part matters and you should be able to rename it.

But again, we can simplify a lot what is happening here. I don't think having the ansible scripts handle the keystore and truststore is a good idea. We can easily move them to the Hawkular Metrics and Cassandra containers instead and that is more where they belongs.

All the ansible scripts should be doing is creating the certificates and keys, in what format those certificates and keys are imported into the application should be the containers concern.

Ideally the ansible script wouldn't even be generating these certificates and keys at all, the system should be creating them for us. The only problem is that we cannot use the system generated certificates because they don't support client based authentication.